infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -93,4 +95,16 @@ public synchronized void setServerEpoch(long serverEpoch) {
         //TODO need to figure out if we can remove this redundancy
         serverRouter.setServerEpoch(serverEpoch);
     }
+
+    public synchronized long getTailSegment() {
```

Comment:
why do you need this synchronized. I remember the datastore gave you that. I might be wrong.